test/IRGen/weak_import_native_hoist.swift

```diff
@@ -0,0 +1,90 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-frontend -emit-module -emit-module-path %t/weak_import_native_hoist_helper.swiftmodule -parse-as-library %S/Inputs/weak_import_native_hoist_helper.swift -enable-library-evolution
+//
+// RUN: %target-swift-frontend -disable-type-layout -primary-file %s -I %t -emit-ir | %FileCheck %s
+
+// UNSUPPORTED: OS=windows-msvc
+
+import weak_import_native_hoist_helper
+
+// We should not hoist the metadata accessor accross the version check.
```

Comment:
This test case is purely a copy paste of the original lines from `weak_import_native.swift` with the mangled names updated.